addons/point_of_sale/static/src/js/custom_hooks.js

```diff
@@ -172,5 +173,39 @@ odoo.define('point_of_sale.custom_hooks', function (require) {
         };
     }
 
-    return { useErrorHandlers, useAutoFocusToLast, onChangeOrder, useBarcodeReader, useAsyncLockedMethod };
+    function useValidateCashInput(inputRef, startingValue) {
+        const cashInput = useRef(inputRef);
+        const current = Component.current;
+        const decimalPoint = current.env._t.database.parameters.decimal_point;
+        const thousandsSep = current.env._t.database.parameters.thousands_sep;
+        // Replace the thousands separator and decimal point with regex-escaped versions
+        const escapedThousandsSep = escapeRegExp(thousandsSep);
+        const scapedDecimalPoint = escapeRegExp(decimalPoint);
+        function isValidFloat(inputValue) {
+            const floatRegex = new RegExp(`^-?(\\d{1,3}(${escapedThousandsSep}\\d{3})*|\\d*)(${scapedDecimalPoint}\\d*)?$`);
```

Comment:
```suggestion
            const floatRegex = new RegExp(`^-?\\d*(${escapedDecimalPoint}\\d*)?$`);
```

Remove the thousands separator validation. It's not so easy to take it into account. For example in india they have this: `1,23,45,678`. It should be valid but because we are assuming thousand separators should be at every 3 digits, then we invalidate other number systems. Let's go with the simpler validation where the decimal separator is in between 2 sets of digits.